Allow links to be forced to lowercase

rhysgodfrey opened the pull request:
Update the links plugin to allow all local links in content to be forced to lowercase.

Defaults to off (to ensure compatibility with current installations of the plugin).

Only updates local links (regardless of it being an absolute url (/whatever/), full url to the same domain (http://mycommunity.com/whatever/), or protocol relative URL (//mycommunity.com/whatever)).

Doesn't alter external links (links with a different host segment).

Doesn't alter Query string values or Anchors, these can be case sensitive so forcing them to lower case could break links.

rhysgodfrey committed:
Allow links to be forced to lowercase

Update the links plugin to allow all local links in content to be forced to lowercase.

Defaults to off (to ensure compatibility with current installations of the plugin).

Only updates local links (regardless of it being an absolute url, full url to the same domain, or protocol relative URL).

RJN368 commented:
Could we improve performance by only running the anchor link regular expression once?

rhysgodfrey committed:
Refactoring to allow the regex to only be called once

Have to pass around the string value and replace into that, to ensure all transforms are applied.
